client/lib/human-date/index.js

```diff
@@ -52,3 +49,27 @@ export default function humanDate(
 
 	return dateOrMoment.format( dateFormat );
 }
+
+export function useHumanDate( dateOrMoment, dateFormat, interval = EVERY_TEN_SECONDS ) {
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+	const [ humanDate, setHumanDate ] = useState(
+		getHumanDateString( dateOrMoment, dateFormat, moment, translate )
+	);
+
+	useInterval( () => {
+		const newHumanDateString = getHumanDateString( dateOrMoment, dateFormat, moment, translate );
+		if ( humanDate !== newHumanDateString ) {
+			setHumanDate( newHumanDateString );
+		}
+	}, interval );
+
+	useEffect( () => {
+		const newHumanDateString = getHumanDateString( dateOrMoment, dateFormat, moment, translate );
+		if ( humanDate !== newHumanDateString ) {
+			setHumanDate( newHumanDateString );
+		}
+	}, [ setHumanDate, humanDate, dateOrMoment, dateFormat, moment, translate ] );
+
+	return humanDate;
+}
```

Comment:
The hook can be written like this:
```js
const [ tick, doTick ] = useReducer( t => t + 1, 0 );
useInterval( doTick, interval );
return useMemo( () => {
  return getHumanDate( date, format, moment, translate );
}, [ tick, date, format, moment, translate ] );
```
The `useInterval` merely increments a counter, and the counter value is one of the dependency values that makes the result to be recomputed.

Reply:
👍🏻 implemented in https://github.com/Automattic/wp-calypso/pull/57826/commits/e8853ac1ed502377447ea0fc6603b6e09499ae00